src/transformStateWithClones.js

```
@@ -1,13 +1,38 @@
 'use strict';
-
 /**
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const contagem = [];
+
+  for (const n of actions) {
+    const lastState =
+      contagem.length > 0 ? contagem[contagem.length - 1] : state;
+    let newState = { ...lastState };
+
+    switch (n.type) {
+      case 'addProperties':
+        Object.assign(newState, n.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of n.keysToRemove) {
+          delete newState[key];
+        }
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
+    }
```

This violates checklist item #3: "switch/case should always have default case for error handling." 

Your `switch (n.type)` (lines 17–31) currently has cases for `'addProperties'`, `'removeProperties'`, and `'clear'`, but no `default`. Add a default branch that explicitly handles unknown action types (for example, throw an Error with the unknown type) so invalid actions are not silently ignored. Example:

```javascript
default:
  throw new Error(`Unknown action type: ${n.type}`);
```

Insert the default case inside the switch; keep the single `contagem.push(newState)` after the switch as implemented.